### ISSUE TITLE
SSH connection to ipv6 

### ISSUE THREAD

**Issue by WolfgangDpunkt** [cronicle-edge/cronicle-edge#139]
The URI format required for the sshx plugin does not yet support ipv6, does it?

`ssh://root:password@2001:db8::1428:57ab:22662`
(Port = 22662)

It seems not possible to specify a port without conflicting with the IP address format. Even specifying the IP in [ ] does not help.

Is there a workaround?
Thanks :-)

**Comment by mikeTWC1984** [comment]
Check if this will work:
ssh://root:password@[2001:db8::1428:57ab]:22662

**Comment by mikeTWC1984** [comment]
What error you get when using []. I see it get parsed properly. Perhaps ssh package doesn't like brackets though

![image](https://github.com/user-attachments/assets/87efffe2-49e9-468f-9d54-0b501b494567)

**Comment by WolfgangDpunkt** [comment]
Hello, thank you for your answer. I have tried it in exactly these format and unfortunately I still get the error:

Error: getaddrinfo ENOTFOUND [2001:db8::1428:57ab]

**Comment by WolfgangDpunkt** [comment]
How can I check if my ssh package  doesn't like brackets?

**Comment by mikeTWC1984** [comment]
Can you connect via ssh in bash? Like ```ssh root@[2001:db8::1428:57ab:22662]``` ?  Perhaps that's firewall issue. I have some machines with openssh, but fail to connect via ipv6 interface (using ssh cli)

**Comment by mikeTWC1984** [comment]
actually ssh doesn't seem to like brackets. Just try ```ssh root@2001:db8::1428:57ab -p 22662``` . In my case it connects when on that remote machine, but cannot connect externally.

**Comment by WolfgangDpunkt** [comment]
It does indeed not work with brackets, but `ssh root@2001:db8::1428:57ab -p 22662` works in the bash of the host server. 
However, cronicle is running in a docker container (cronicle/edge:latest)...maybe this alpine contains a ssh version that is not compatible?
I tried `docker exec cronicle-edge ssh` and got the error message `not found in $PATH`. so I installed it `docker exec cronicle-edge apk add ssh` and when I try now
`docker exec ssh root@2001:db8::1428:57ab -p 22662 `
I get `Network unreachable`.

A ping to the ipv6 address via the container also fails with the same error message, an ipv4 address works.

Apparently the error is not due to cronicle but to the fact that the container has no ipv6 connection, I will investigate how I can fix this. Thank you for your support in troubleshooting.

**Comment by WolfgangDpunkt** [comment]
The good news is that it was very easy to teach the Docker container ipv6:
https://docs.docker.com/engine/daemon/ipv6/

(Perhaps this can be included in the Dockerfile as standard?)

The not so good news, unfortunately there is a new error.
The difficulty remains that you cannot specify the port because it cannot be separated with `:` for an ipv6 address.
Unfortunately, the trick with square brackets does not work. I think something needs to be adapted to the plugin so that the syntax is also possible for ipv6.

```
node:internal/url:775
    this.#updateContext(bindingUrl.parse(input, base));
                                   ^

TypeError: Invalid URL
    at new URL (node:internal/url:775:36)
    at Object.<anonymous> (/opt/cronicle/bin/sshx-plugin.js:41:72)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'ERR_INVALID_URL',
  input: 'ssh://root@2001:db8::1428:57:22662'
}

Node.js v20.11.1
```

**Comment by mikeTWC1984** [comment]
Yeah, I did some testing, the underlying npm module (ssh2) just don't like brackets. Other than that it seem to work with no issue. The fix should very simple (just trim [/] from the hostname), will release it soon. I believe just need to update this line, if you want you can play with it.
https://github.com/cronicle-edge/cronicle-edge/blob/479d6c0a01c95d6c7b86d74ae3fd5dbccb9f69df/bin/sshx-plugin.js#L103
